addons/microsoft_calendar/models/calendar.py

```diff
@@ -74,13 +75,41 @@ def create(self, vals_list):
         if self._check_microsoft_sync_status() and not notify_context and recurrency_in_batch:
             self._forbid_recurrence_creation()
 
-        # for a recurrent event, we do not create events separately but we directly
-        # create the recurrency from the corresponding calendar.recurrence.
-        # That's why, events from a recurrency have their `need_sync_m` attribute set to False.
-        return super(Meeting, self.with_context(dont_notify=notify_context)).create([
-            dict(vals, need_sync_m=False) if vals.get('recurrence_id') or vals.get('recurrency') else vals
-            for vals in vals_list
-        ])
+        # Create events separately since the sender user may differ between events.
+        events = self.env['calendar.event']
+        users_by_id = self._get_users_by_id(vals_list)
+        vals_by_user = defaultdict(lambda: [])
+        for vals in vals_list:
+            # If event has a different organizer, check its sync status and verify if the user is listed as attendee.
+            sender_user, partner_ids = self._get_organizer_user_change_info(vals, users_by_id)
+            partner_included = partner_ids and len(partner_ids) > 0 and sender_user.partner_id.id in partner_ids
+            self._check_organizer_validation(sender_user, partner_included)
+            # Group new events values by user for later batch creation.
+            vals_by_user[sender_user].append(vals)
+            # Events from a recurrency have their `need_sync_m` attribute set to False.
+            if vals.get('recurrence_id') or vals.get('recurrency'):
+                vals['need_sync_m'] = False
+
+        for sender_user in vals_by_user:
+            events |= super(Meeting, self.with_user(sender_user).with_context(dont_notify=notify_context)).create(vals_by_user[sender_user])
+        return events
+
+    def _check_organizer_validation(self, sender_user, partner_included):
+        """ Check if the proposed event organizer can be set accordingly. """
+        if sender_user and sender_user != self.env.user:
+            if not self.with_user(sender_user)._check_microsoft_sync_status():
+                raise ValidationError(
+                    _("For having a different organizer in your event, it is necessary that "
+                      "the organizer have its Odoo Calendar synced with Outlook Calendar."))
+            elif not partner_included:
+                raise ValidationError(
+                    _("It is necessary adding the proposed organizer as attendee before saving the event."))
+
+    def _get_users_by_id(self, vals_list):
+        """ Load users of parameters list from the database. """
+        user_ids = [vals['user_id'] for vals in vals_list if vals.get('user_id')]
+        users = self.env['res.users'].browse(user_ids)
+        return {user.id: user for user in users}
```

Comment:
I think this method is useless. You have the information of the user inside the vals. see my commit.